realm/realm-library/src/objectServer/java/io/realm/mongodb/mongo/events/BaseChangeEvent.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.mongodb.mongo.events;
+
+import org.bson.BsonDocument;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the set of properties that exist on all MongoDB realm change events produced
+ * by watch streams in this SDK. Other change event types inherit from this type.
+ *
+ * @param <DocumentT> The type of the full document in the change event.
+ */
+public abstract class BaseChangeEvent<DocumentT> {
+    @Nonnull
```

Comment:
`@Nonnull` annotations shouldn't be required. Find a `package-info.java` and copy it to `io.realm.mongodb.mongo.events`. Then remove all `@Nonnull`s.